### PR TITLE
migrate to apollo-modules-node and backend-modules

### PR DESCRIPTION
**Pull request by patte**
move all auth and user related graphql queries and resolvers
and middleware to backend-modules-auth.
move document graphql queries and resolvers to backend-module-documents.
require local graphql with apollo-modules-node
don't yet depend backend-modules-base, which handles all the express, graphql and
database heavy lifting, but use a copy of it's code locally.

TODOs
- [x] migrate test setup
- [x] solve npm import errors
- [x] use published npm packages not local links

### CODE REVIEW

**Comment by patte** [comment]
@tpreusse @lukasbuenger could you have a quick look at this and [backend-modules](https://github.com/orbiting/backend-modules)? The diff is maybe too crazy, but I would appreciate if you could check out the overall setup and structure.

**Review by lukasbuenger — APPROVED** [review]
I already got my hands on it this morning, when I had to fix the embed union. 
Doesn't look too different from what we had before, except of course the outsourcing of certain parts. I'm ok with it.

**Comment by lukasbuenger** [comment]
One thing though:
I already fixed this but adding new .env keys or making them mandatory has to have at least some impact on the README. 
Also, having a little more context on what 3rd party stuff you use would help, especially if it's as pivotal as e.g. graphql-tools.

File: graphql/resolvers/_mutations/commit.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ module.exports = async (_, args, { pgdb, req, user, t, pubsub }) => {
     })
     // auto_init seems to be async and takes some time
     // we sporadically saw: Git Repository is empty. on commit
-    await sleep(1000)
+    await sleep(3000)
```
Comment (tpreusse) [inline graphql/resolvers/_mutations/commit.js]: Off topic: could we change this to a api call that verifies this every 500ms? With a fail timeout of ~30s?

Quick fix: only increase to 3000ms in test env.

**Review by tpreusse — COMMENTED** [review]
Generally looks good. But I haven't read line by line.

I hope the subscription issue only affects the tests. And notice an off topic issue, the decreasing create performance.

File: graphql/resolvers/_mutations/commit.js (hunk 1 above)
Comment (patte) [reply graphql/resolvers/_mutations/commit.js]: thank's for the hint! I implemented polling. timeout: 6s should be more than enough IMHO.

**Comment by lukasbuenger** [comment]
Tiger!

**Pull request merged**